### PR TITLE
Regression test for array of struct with a single field name "element" in Parquet

### LINKED ISSUE #8694: Add test for Parquet schema interpretation problem described in SPARK-16344
Apache JIRA: https://issues.apache.org/jira/browse/SPARK-16344.

Refer to [`ParquetHiveCompatibility.scala#139`](https://github.com/apache/spark/blob/85e252e8503534009f4fb5ea005d44c9eda31447/sql/hive/src/test/scala/org/apache/spark/sql/hive/ParquetHiveCompatibilitySuite.scala#L139).

A Parquet file with a column of the following type needs to be read by `spark-rapids` in the same way as in Apache Spark:
```sql
ARRAY<STRUCT<element:BIGINT>>
```

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds a regression test for a previous bug in Spark 1.6. This is a weird corner case. Users may hit this issue if they have a schema that has a list of structs with a single element field named `element`

Details for it can be found at #8694 

fixes #8694

### CODE REVIEW

File: integration_tests/src/main/python/parquet_write_test.py [hunk 1]
```diff
@@ -755,3 +755,13 @@ def test_write_with_planned_write_enabled(spark_tmp_path, planned_write_enabled,
         lambda spark, path: spark.read.parquet(path),
         data_path,
         conf)
+
+# Issue to test a known bug https://github.com/NVIDIA/spark-rapids/issues/8694 to avoid regression
+@ignore_order
+def test_write_list_struct_single_element(spark_tmp_path):
```
Comment (jlowe) [inline integration_tests/src/main/python/parquet_write_test.py:761]: Either there should be a similar test in parquet_test to test the GPU read side or this should be extended to test the read on the GPU.  This is only testing GPU write, but the CPU performs both reads.  To add the test of the GPU read here, could add something like:
```
  cpu_path = data_path + "/CPU"
  assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.read.parquet(cpu_path), conf)
```

**Review by jlowe — DISMISSED** [review]

**Comment by jlowe** [comment]
build

**Review by revans2 — DISMISSED** [review]

**Review by jlowe — DISMISSED** [review]

**Comment by jlowe** [comment]
build

**Comment by razajafri** [comment]
merge conflict caused the review to be revoked. @jlowe PTAL

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**